src/Model/CachingIterator.php

```diff
@@ -83,12 +88,13 @@ public function current()
     {
         $currentItem = current($this->items);
 
-        return $currentItem !== false ? $currentItem[self::FIELD_VALUE] : false;
+        return $currentItem !== false ? $currentItem[self::FIELD_VALUE] : null;
```

Comment:
This is something that surprised me when I worked on this class #1118.

The return value is not specified. `ArrayIterator` returns `null` while `EmptyIterator` [throws an exception](https://www.php.net/manual/en/emptyiterator.current.php).

The fix looks good.

Reply:
The class originally used [`current()`](https://www.php.net/current), which returns `false`. That was preserved when it was later refactored in [3049b5d](https://github.com/mongodb/mongo-php-library/commit/3049b5dba3fd0ef5b95a5f2d12a140a8b6a2a180#diff-be61b04c89f7e9634e8b3d05aede2915bdbadefd72995eb1ecee98bcae3eb420L90-R94).

That said, I have no object to being consistent with ArrayIterator and returning `null`.